shared/src/simulation/activities/provide-personnel-from-vehicles.ts

```diff
@@ -0,0 +1,210 @@
+import { IsString, IsUUID } from 'class-validator';
+import type { SimulatedRegion } from '../../models';
+import type { PersonnelType } from '../../models/utils';
+import {
+    PersonnelResource,
+    VehicleResource,
+    getCreate,
+} from '../../models/utils';
+import {
+    addResourceDescription,
+    greaterEqualResourceDescription,
+    ResourceDescription,
+} from '../../models/utils/resource-description';
+import type { ExerciseState } from '../../state';
+import {
+    cloneDeepMutable,
+    StrictObject,
+    UUID,
+    uuidArrayValidationOptions,
+    uuidValidationOptions,
+} from '../../utils';
+import { IsValue } from '../../utils/validators';
+import { IsResourceDescription } from '../../utils/validators/is-resource-description';
+import { ResourceRequiredEvent } from '../events';
+import { sendSimulationEvent } from '../events/utils';
+import { nextUUID } from '../utils/randomness';
+import type {
+    SimulationActivity,
+    SimulationActivityState,
+} from './simulation-activity';
+import type { UnloadVehicleActivityState } from './unload-vehicle';
+
+export class ProvidePersonnelFromVehiclesActivityState
+    implements SimulationActivityState
+{
+    @IsValue('providePersonnelFromVehicleActivity' as const)
+    public readonly type = 'providePersonnelFromVehicleActivity';
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly id: UUID;
+
+    @IsResourceDescription()
+    public readonly requiredPersonnelCounts: ResourceDescription<PersonnelType>;
+
+    @IsUUID(4, uuidArrayValidationOptions)
+    public readonly vehiclePriorities: readonly UUID[];
+
+    @IsString()
+    public readonly key: string;
+
+    /**
+     * @deprecated Use {@link create} instead
+     */
+    constructor(
+        id: UUID,
+        requiredPersonnelCounts: ResourceDescription<PersonnelType>,
+        vehiclePriorities: UUID[],
+        key: string
+    ) {
+        this.vehiclePriorities = vehiclePriorities;
+        this.requiredPersonnelCounts = requiredPersonnelCounts;
+        this.id = id;
+        this.key = key;
+    }
+
+    static readonly create = getCreate(this);
+}
+
+// Because this activity relies on a cancel condition, we cannot model it as a DelayEventActivity
```

Comment:
Is this comment still correct or was it only valid for an intermediate version of this activity?

Reply:
It was left after I copied the boilerplate from another activity. Removed it in https://github.com/hpi-sam/digital-fuesim-manv/pull/850/commits/6da4a17c8713e5a0428c67dcd382e3718551befb